styles/_mixins.scss

```diff
@@ -0,0 +1,12 @@
+@mixin respond-to($breakpoint) {
+    
+    @if $breakpoint == sm {
+        @media (max-width: $breakpoint-sm) {
+            @content;
+        }
+    } @else if $breakpoint == lg {
+        @media (min-width: $breakpoint-lg) {
+            @content;
+        }
+    }
+}
```

Comment:
I saw your discussion in slack with @fraserclark on this.
I actually already created a bunch of responsive helpers for this in; `@/components/responsive`.

Regarding @fraserclark 's point on not wanting to use names like "tablet", I'm not very opinionated on this, but would be interested to hear the reasoning behind it. My only comment regarding this would be that I normally prefer names that I can recognize, and don't have to go look up what the "shortcode" for the size is.

Reply:
@andrewebdev and @fraserclark, I went ahead and switched to using the breakpoint labels in responsive.scss. Honestly, I totally forgot you had added those breakpoints already Andre. Do you think we should move that file out of components and into the "/styles" directory? It seems to make more sense to me than keeping it in components.

Reply:
yeah you can move it to where it makes more sense. I don't recall my reasoning why I put it in that directory to begin with, but you are right it doesn't make sense there, and is probably why you didn't realize it's there already ;)

Reply:
Thanks @andrewebdev. I moved it under "/styles"

Reply:
sm, md, lg, xl, xxl, you create a more flexibility, and I think maintainable CSS system. 

It’s based on the fundamental that the viewport size rather some kind of "device", which is often ambiguous and overlapping categories of devices. We can purely focus on the dimension of width and slightly abstracting the device.

mobile screens can range from 320px to 600px, tablets from 500px to 1400px, and laptops/desktops from 1100px to 4K and beyond. There significant overlap and devices have ambiguity - Like is a 2-in-1 laptop in tablet mode a "tablet"? What about a large phone?

Also they allow helpful responsive utilities to be formed based on the same break points - classic examples are like 

Padding: py-3 for tight padding on small screens, py-md-5 for increased padding on medium screens, and py-lg-3 to reduce padding again on large screens. - This is used extensively in Bootstrap and Tailwind (and most frameworks)

Visibility:  like d-none d-md-block allow you to hide an element on smaller screens and display it on medium or larger screens. These classes become quite intuitive and means we don't need every component having its own logic etc

There are also other utitlies like flex direction, flex-row/col, css grids which often are suited to "utilities" 

In device targeting, we can use more specific heuristics if we need to figure out where we don't want hover states or whatever - but its an "add-on" etc

Reply:
@fraserclark, I definitely agree with you that device widths are ambiguous nowadays. We can see that by looking at statcounter's width ranges for devices worldwide: https://gs.statcounter.com/screen-resolution-stats/desktop/worldwide#monthly-202307-202407-bar

The terms used for the breakpoints is subjective, and should generally be guided by what the team finds easier to use. I am open to either sets of terms. Also, the breakpoints used in "responsive.scss" are definitely not final. My understanding is that we will be developing for `fluid layout` first, and then the breakpoints we decide to use will be less about device widths, and more about where they make sense for our design.  Based on that, it does seem like using more general terms like "`xs, sm, md, lg, xl`" would make sense. @andrewebdev and @fraserclark, if we are all in agreement, I can update the terms used in responsive.scss for now. If it becomes difficult to work with, we can revisit.